app/controllers/tasks_controller.rb

```diff
@@ -85,8 +85,10 @@ def add_to_collection
   end
 
   def download
-    zip_file = ProformaService::ExportTask.call(task: @task)
+    zip_file = ProformaService::ExportTask.call(task: @task, options: {version: params[:version]})
     send_data(zip_file.string, type: 'application/zip', filename: "task_#{@task.id}.zip", disposition: 'attachment')
+  rescue ProformaXML::PostGenerateValidationError => e
+    redirect_to :root, danger: JSON.parse(e.message).map {|msg| t("proforma_errors.#{msg}", default: msg) }.join('<br>')
```

Comment:
What about redirecting to `@task`? I feel, this would increase usability.

Reply:
I wasn't sure what the optimal way would be. Currently (on master) when you click the link you change your current location to start the download, but the task page stays loaded. Therefore I added the `target: '_blank'` to the `link_to`'s. This works well in the success case - a new tabs opens, the download starts and the tab closes again.
In the error case the tab doesn't close and it felt wrong to just redirect back to the task. The obvious answer would be to use ajax to start the download and show the error, but I think that's and over-engineered approach for such an edge case. The user should only get that error if they start messing around with the parameters (like you :) )

Reply:
Ah, that makes sense -- I didn't notice that behavior in the first place (since I was opening the link in a new tab anyway). Thank that's fine for me.